__tests__/reactTags.test.js

```diff
@@ -589,3 +590,38 @@ test('should allow duplicate tags when allowUnique is false', () => {
     text: 'Apple',
   }]);
 });
+
+[
+  { allowEmpty: true, title: 'allow empty tags' },
+  { allowEmpty: false, title: 'not allow empty tags' },
+].forEach((data) => {
+  const { title, allowEmpty } = data;
+  test(`should ${title} when allowEmpty is ${allowEmpty}`, () => {
+    const actual = [];
+    const $el = mount(
+      mockItem({
+        handleAddition(tag) {
+          actual.push(tag);
+        },
+        allowEmpty: allowEmpty,
+      })
+    );
+
+    expect($el.instance().props.tags).to.have.deep.members(defaults.tags);
+    const $input = $el.find('.ReactTags__tagInputField');
+    $input.simulate('change', { target: { value: '' } });
+    $input.simulate('keyDown', { keyCode: ENTER_ARROW_KEY_CODE });
+    if (allowEmpty) {
```

Comment:
noooo lets not have if else conditions for assertions inside specs. This approach is fine when the assertions are exactly same. Please create two different specs